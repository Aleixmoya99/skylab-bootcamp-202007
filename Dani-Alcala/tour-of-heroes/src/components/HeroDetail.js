import React from 'react';

class HeroDetail extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			heroName: 'Bombasto',
			heroId: 14,
		};
		this.onFieldChange = this.onFieldChange.bind(this);//le asigno el contexto que me interesa, que es el objeto que se crea con esta clase. El form tiene su propio contexto
	}
	onFieldChange(event) {
		this.setState({
			// heroName: event.target.value //es pot fer aixi, més fàcil
			[event.target.name]: event.target.value //propiedad dinamica, així està preparat per si poso un altre input amb el name associat correctament a this.state
		})
		console.log(this.state.heroName)
	}

	render() {
		return (
			<form>
				<p>id: {this.state.heroId}</p>
				<label htmlFor="heroName">
					name:
					<input
						name="heroName"
						placeholder="Hero name"
						value={this.state.heroName}
						onChange={this.onFieldChange}
					/>
				</label>
			</form>
		);
	}
}

export default HeroDetail;
