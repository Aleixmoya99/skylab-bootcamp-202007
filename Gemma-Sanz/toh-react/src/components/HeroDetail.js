import React from 'react';
import heroData from '../heroData';
import '../styles.css';

class HeroDetail extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            heroName: "Bombasto"/* heroData[0].name */,
            heroId: 14
        };
        this.onFieldChange = this.onFieldChange.bind(this);
    }

    onFieldChange(myEvent) {
        this.setState({
            [myEvent.target.name]: myEvent.target.value
        });
    }

    render() {
        return (
            <form className="containerComponent">
                <h2><span>{this.state.heroName}</span> details!</h2>
                <p>
                    id:<span>{this.state.heroId}</span>
                </p>
                <label htmlFor="heroName">
                    name:
                <input
                        name="heroName"
                        placeholder="Hero name"
                        value={this.state.heroName}
                        onChange={this.onFieldChange} />
                </label>
            </form>
        )
    }
}

export default HeroDetail;