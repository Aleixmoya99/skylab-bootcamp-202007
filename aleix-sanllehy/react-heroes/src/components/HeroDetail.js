import React from 'react';
import heroData from './hero.mock';

class HeroDetail extends React.Component {
	constructor(props) {
		super(props);
		this.state = this.getId();
		//this.state = { heroId: null, heroName: '' };
		this.onFieldChange = this.onFieldChange.bind(this);
	}

	onFieldChange(myEvent) {
		this.setState({ name: myEvent.target.value });
		//this.setState({ [myEvent.target.name]: myEvent.target.value });
		//console.log(myEvent.target.name, '=>', myEvent.target.value);
	}

	getId() {
		const url = window.location.pathname;
		const urlId = +url.split('/')[2];
		return heroData.find((element) => element.id === urlId);
	}

	render() {
		return (
			<main className="hero-details">
				<h3>
					<span>{this.state.name}</span> details!
				</h3>
				<form>
					<p className="hero-details__element">
						<span className="hero-details__id">id:</span>
						<span className="hero-details__id">{this.state.id}</span>
					</p>
					<label className="hero-details__name" htmlFor="heroName">
						name:
						<input
							name="heroName"
							placeholder="Hero name"
							value={this.state.name}
							onChange={this.onFieldChange}
						/>
					</label>
				</form>
				<span className="id btn btn-secondary">Back</span>
			</main>
		);
	}
}

export default HeroDetail;
