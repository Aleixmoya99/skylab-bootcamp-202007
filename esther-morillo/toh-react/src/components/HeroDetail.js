import React from 'react';
import heroList from '../hero.mock';
import heroStore from '../stores/heroStore';
import { Prompt, Link } from 'react-router-dom';

class HeroDetail extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			heroName: '',
			heroId: null,
			formIsDirty: true
		};

		this.onFieldChange = this.onFieldChange.bind(this);
	}

	onFieldChange(myEvent) {
		this.setState({
			[myEvent.target.name]: myEvent.target.value,
		});
	}

	

	getHeroById(id) {
		return heroList.find((hero) => hero.id === id);
	}

	componentDidMount() {
		const hero = this.getHeroById(+this.props.match.params.heroId);
		this.setState({
			heroName: hero.name,
			heroId: hero.id
		});
	}

	render() {
		return (
			<>
			<form>
				<p>id: {this.state.heroId}</p>
				<label htmlFor="heroName">
					name:
					<input
						name="heroName"
						placeholder="Hero name"
						value={this.state.heroName}
						onChange={this.onFieldChange}
					/>
				</label>
				<Prompt
					when={this.state.formIsDirty}
					message="Are you sure you want to navigate away?"
				/>
			</form>

			<Link 
			to={`/hero/`}
			onClick={this.getHeroNewName}
			>
					Back
			</Link>
			</>
		);
	}
}

export default HeroDetail;
