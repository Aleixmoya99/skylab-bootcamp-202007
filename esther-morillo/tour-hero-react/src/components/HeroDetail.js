import React from 'react'


function HeroDetail({ id, name }) {
    return (
        <>
            <div>
                <h2>{name} details!</h2>
                <p>id:<span>{id}</span></p>
                <label>name:<input></input></label>
            </div>
        </>
    );
}

export default HeroDetail;