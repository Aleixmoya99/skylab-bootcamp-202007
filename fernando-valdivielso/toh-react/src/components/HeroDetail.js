import React from 'react';
import heroData from '../heroData';
import { Prompt } from 'react-router-dom'

class HeroDetail extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			heroName: '',
			heroId: null,
			formIsDirty: false
		};

		this.onFieldChange = this.onFieldChange.bind(this);
	}

	componentDidMount() {
		const heroe = this.findHeroById(+this.props.match.params.heroId)
		this.setState({
			heroName: heroe.name,
			heroId: heroe.id
		})
	}

	findHeroById(heroId) {
		return heroData.find((hero) => hero.id === heroId)

	}

	onFieldChange(myEvent) {
		this.setState({
			[myEvent.target.name]: myEvent.target.value, //[myEvent.target.name] = heroName 
			formIsDirty: true

		});
	}

	render() {
		return (
			<form>
				<p>{this.state.heroName} details!</p>
				<p>id: {this.state.heroId}</p>
				<label htmlFor="heroName">

					name:
					<input
						name="heroName"
						placeholder="Hero name"
						value={this.state.heroName}
						onChange={this.onFieldChange}
					/>

				</label>
				<Prompt
					when={this.state.formIsDirty}
					message='Are you sure you wamt to exit this page?'
				/>
			</form>
		);
	}
}

export default HeroDetail;
