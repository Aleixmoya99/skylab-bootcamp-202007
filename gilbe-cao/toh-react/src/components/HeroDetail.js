import React from 'react';
import heroList from '../hero.mock';
import { Prompt } from 'react-router-dom';

class HeroDetail extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			heroName: '',
			heroId: null,
			isFormDirty: false
		};

		this.onFieldChange = this.onFieldChange.bind(this);
	}

	onFieldChange(myEvent) {
		this.setState({
			[myEvent.target.name]: myEvent.target.value,
			isFormDirty: true
		});
	}

	getHeroById(id) {
		return heroList.find((hero) => hero.id === id);
	}

	componentDidMount() {
		const hero = this.getHeroById(+this.props.match.params.heroId);
		this.setState({
			heroName: hero.name,
			heroId: hero.id
		});
	}

	render() {
		return (
			<form>
				<p>id: {this.state.heroId}</p>
				<label htmlFor="heroName">
					name:
					<input
						name="heroName"
						placeholder="Hero name"
						value={this.state.heroName}
						onChange={this.onFieldChange}
					/>
				</label>
				<Prompt
					when={this.state.isFormDirty}
					message="Are you sure you want to navigate away?"
				/>
			</form>
		);
	}
}

export default HeroDetail;
