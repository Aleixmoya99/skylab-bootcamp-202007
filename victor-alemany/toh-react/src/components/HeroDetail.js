import React, { useState, useEffect } from 'react';
import heroStore from '../stores/heroStore';
import { getHero } from '../actions/heroActions';

function HeroDetail(props){
    const [heroes, setHeroes]= useState([]);

    useEffect(()=>{
        heroStore.addChangeListener(onChange);
        if(heroes.length === 0) getHero();
        return ()=> heroStore.removeChangeListener(onChange);
    },[]);

    function onChange(){
        const id = props.match.params.heroId
        setHeroes(heroStore.getHeroById(id));
    }

    useEffect(()=>{
        heroStore.addChangeListener(onChange);
        if(heroes.length === 0) updateHero();
        return ()=> heroStore.removeChangeListener(onChange);
    },[]);

    function onChange(){
        const id = props.match.params.heroId
        setHeroes(heroStore.getHeroById(id));
    }

		return (
			<form>
                <div>
                    <h2>{heroes.name} details</h2>
                </div> 
				<p>id: {heroes.id}</p>
				<label htmlFor="heroName">
					name:
					<input
						name="heroName"
						placeholder="Hero name"
						value={heroes.name}
                        onChange={this.onFieldChange}/>
				</label>
			</form>
            
        );	
}

export default HeroDetail;