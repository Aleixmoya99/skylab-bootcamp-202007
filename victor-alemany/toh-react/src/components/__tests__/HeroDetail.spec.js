import renderer from 'react-test-renderer'
import HeroDetail from './../HeroDetail'
import React from 'react'
import { BrowserRouter } from 'react-router-dom'


describe('Hero Detail snapshot', ()=>{
    const  props = {
        match: {
            params: {
                heroId: 14
            }
        }
    };
    const treeDetail = renderer.create(
     <BrowserRouter>
        <HeroDetail {...props} />
     </BrowserRouter>
    );

    it('should match', ()=>{
        expect(treeDetail).toMatchSnapshot();
    })
})
