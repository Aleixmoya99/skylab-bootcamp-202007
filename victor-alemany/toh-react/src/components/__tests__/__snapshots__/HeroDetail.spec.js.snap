// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Detail snapshot should match 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Register a new hero:
  </h2>
  <input
    name="heroName"
    onChange={[Function]}
    placeholder="Hero name"
    value=""
  />
  <p>
    <button
      type="submit"
    >
      Save Hero
    </button>
  </p>
</form>
`;
